### PR TITLE
Fix malformed scheduler test case

### PR DESCRIPTION
**Pull request by mihails-strasuns-sociomantic**
- Fix of test case which was presumably causing problems for @ben-palmer-sociomantic 
- Unrelated fix of `task_pending_cleanup` which I noticed while investigating

### CODE REVIEW

**Comment by mihails-strasuns-sociomantic** [comment]
@ben-palmer-sociomantic can you check this please?

**Comment by ben-palmer-sociomantic** [comment]
> @ben-palmer-sociomantic can you check this please?

Yes! fixes my issue, thanks!

**Comment by mihails-strasuns-sociomantic** [comment]
Ping @gavin-norman-sociomantic

File: src/ocean/task/Task.d [hunk 1]
```diff
@@ -462,7 +462,16 @@ public abstract class Task : ISuspendable
             Task.task_pending_cleanup = this;
             this.fiber.yieldAndThrow(e);
             Task.task_pending_cleanup = null;
-            return true;
+
+            // if task was resumed with `kill` from 'pending cleanup' state
+            // just quit the method as if it was originally terminated by kill
+            if (this.to_kill)
+            {
+                debug_trace("<{}> termination (killed)", cast(void*) this);
```
Comment (gavin-norman-sociomantic) [inline src/ocean/task/Task.d:471]: Just to check: the `to_kill` flag should remain set to true, in this case?

Comment (mihails-strasuns-sociomantic) [reply src/ocean/task/Task.d:471]: Yes, but looks like my other fix to reset flags in `assignTo` method was only merged into later branch. I will add it here too.

Comment (mihails-strasuns-sociomantic) [reply src/ocean/task/Task.d:471]: Updated

**Review by gavin-norman-sociomantic — APPROVED** [review]

**Pull request merged**